### ISSUE TITLE
"Hide boosts" appears not to be working

### ISSUE THREAD

**Issue by debbiegoldsmith** [tootsuite/mastodon#8172]
I follow Eugen on mastodon.social, but I've had his account muted due to volume. I noticed the "Hide boosts from" and "Show boosts from" feature in the profile, and tried unmuting Eugen but hiding boosts.

I'm still seeing boosts from Eugen in my timeline. The feature appears not to have any effect. Either that or I'm misinterpreting how it's supposed to work. The status in the profile is updating (i.e., it flips between "hide boosts" and "show boosts") but boosts seem to appear regardless.

My account is also on mastodon.social, dgoldsmith.

I did do a cursory search for the word "boost" in open issues, but didn't see anything.

* * * *

- [ x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by konomikitten** [comment]
I just came across this issue on mastodon.online as well, this bug is getting pretty old is the hide boosts feature ever going to work?

**Comment by virtadpt** [comment]
Will this feature ever be added to the UI?  Please?  This is starting to turn into a problem.

**Comment by sparky005** [comment]
Just commenting to say that this appears to still be a problem, two years later :( are we just misunderstanding how this option is supposed to work?

**Comment by trwnh** [comment]
the way it works is it hides anything boosted *by* them, not *from* them

**Comment by sparky005** [comment]
Understood, thanks for clarifying. I think part of the confusion is that clicking "hide boosts" doesn't visually appear to do anything. I still see all boosts on the user's page and all boosts in my home timeline up until that point. It doesn't seem that the option does anything retroactively, which is perhaps why it's confusing and seems like it's doing nothing at all.

As far as I can tell, it is indeed actually working, it's just hard to know for sure without being hyper-aware of who is boosting things on my home timeline and making sure it's not any of the users I've turned boosts off for, and it's not really clear if / how it works when looking at the user's page. 

So it's not broken, yeah, but it's definitely confusing for a new user

**Comment by Krinkle** [comment]
Good point on it not being retroactive. That's something that I'd hope could be improved.

It's likely too expensive to do as a runtime filter (given Mastodon goes out of its way by storing posts as a per-person queue), but it could e.g. queue a background job that relatively soon (based on load) goes through your queue and removes entries that would be hidden by your current preferences. Each person's queue is limited to 400 entries so that wouldn't take too long to do I think. I imagine this is only feasible to do reductively, not the other way around. E.g. following someone new or re-enabling boosts isn't going to repopulate your Home with posts originaly rejected at entry or never received.

**Comment by eishiya** [comment]
Unfollowing a user immediately removes them from your home timeline, is that done as a high-priority server-side task, or is it client-side?

Hiding boosts retroactively could perhaps be handled client-side, similarly to how the "show boosts" toggle on the home timeline works. So, if you hide someone boosts, their older boosts would immediately be hidden in the client, and if you show boosts, then any of their old boosts still available on your home timeline would reappear, while any that were dropped while you had their boosts hidden would not.

Most of the time, when I disable boosts from a user, it's because they've flooded my home timeline and I'd like to see other people's posts and boosts, but because it's not retroactive, I can only fix my future home timeline, I can't get rid of the existing flood. I can hide all boosts, but then I'd be unable to see everyone else's boosts too.

**Comment by akkartik** [comment]
@eishiya As a possible workaround, I wonder if unfollow-refollow and immediately hiding boosts will prevent boosts from getting added to your timeline. It does cause some potentially delicate notifications.

**Comment by kefir-** [comment]
I'm guessing "hide boosts" is a federation setting, and not a display filter. If this is the case, and you hide boosts from an account, you just won't receive future boosts. Then it makes sense that you still see the user's boosts when you view their profile, and you still see their old boosts in your timeline.

**Comment by vintprox** [comment]
I think it requires adding a word "further" in "Hide boosts", because it's a bit misleading right now...

**Comment by Squaredude** [comment]
> @eishiya As a possible workaround, I wonder if unfollow-refollow and immediately hiding boosts will prevent boosts from getting added to your timeline. It does cause some potentially delicate notifications.

Possibly things changed since this comment.

A better workaround I've found is to hide boosts, then mute the user, and immediately unmute them.

The timeline becomes retroactively clean and the user is not notified.

Tested on 4.2.1. Not sure how new this method is.

**Comment by YourAutisticLife** [comment]
I have a pull request that aims to fix this. It is really trivial. I've tested it on my instance and AFAIK, it works. Here's the PR:

https://github.com/mastodon/mastodon/pull/28737

If you know of a way to get this integrated into Mastodon, please tell me. I've got four PRs that are languishing, and I'm getting tired of this.

**Comment by bvaughn** [comment]
> > @eishiya As a possible workaround, I wonder if unfollow-refollow and immediately hiding boosts will prevent boosts from getting added to your timeline. It does cause some potentially delicate notifications.
> 
> Possibly things changed since this comment.
> 
> A better workaround I've found is to hide boosts, then mute the user, and immediately unmute them.
> 
> The timeline becomes retroactively clean and the user is not notified.
> 
> Tested on 4.2.1. Not sure how new this method is.

Lovely. This still works.

**Comment by Squaredude** [comment]
> > > @eishiya As a possible workaround, I wonder if unfollow-refollow and immediately hiding boosts will prevent boosts from getting added to your timeline. It does cause some potentially delicate notifications.
> > 
> > 
> > Possibly things changed since this comment.
> > A better workaround I've found is to hide boosts, then mute the user, and immediately unmute them.
> > The timeline becomes retroactively clean and the user is not notified.
> > Tested on 4.2.1. Not sure how new this method is.
> 
> Lovely. This still works.

Indeed, I still use this workaround all the time.

And something I forgot to add, it also works for lists. Especially when using the setting "hide boosts from the home timeline".
